src/content/en/updates/2018/03/smooshgate.md

```diff
@@ -0,0 +1,217 @@
+`Array.prototype.sort = whatever`, it remains non-enumerable.
+
+Currently, `Array.prototype.flatten = mooToolsFlattenImplementation` creates
+an enumerable `flatten` property, so it’s later copied to `Elements`. But if
+we ship a native version of `flatten`, it becomes non-enumerable, and
+_isn’t_ copied to `Elements`. **Any code relying on MooTools’
+`Elements.prototype.flatten` is now broken.**
+
+Although it seems like changing the native `Array.prototype.flatten` to be
+enumerable would fix the problem, it would likely cause even more
+compatibility issues. Every website relying on `for`-`in` to iterate over
+an array (which is a bad practice, but it happens) would then suddenly get
+an additional loop iteration for the `flatten` property.
+
+The bigger underlying problem here is modifying built-in objects. Extending
+native prototypes is generally accepted as a bad practice, as it doesn’t
+compose nicely with other libraries and third-party code. Don’t modify
+objects you don’t own!
+
+## Why don’t we just keep the existing name and break the Web? {: #break-the-web }
+
+In 1996, before CSS became widespread, and long before “HTML5” became a
+thing, [the Space Jam
+website](https://www.warnerbros.com/archive/spacejam/movie/jam.htm) went
+live. Today, the website still works the same way it did 22 years ago.
+
+How did that happen? Did someone maintain that website for all these years,
+updating it every time browser vendors shipped a new feature?
+
+As it turns out, “don’t break the Web” is the number one [design
+principle](https://www.w3.org/TR/html-design-principles/#support-existing-content)
+for HTML, CSS, JavaScript, and any other standard that’s widely used on the
+Web. If shipping a new browser feature causes existing websites to stop
+working, that’s bad for *everyone*:
+
+- visitors of the affected websites suddenly get a broken user experience;
+- the website owners went from having a perfectly-working website to a
+  non-functional one without them changing anything;
+- browser vendors shipping the new feature lose market share, due to users
+  switching browsers after noticing “it works in browser X”;
+- once the compatibility issue is known, other browser vendors refuse to ship
+  it. The feature specification does not match reality ([“nothing but a work
+  of fiction”](https://www.webstandards.org/2009/05/13/interview-with-ian-hickson-editor-of-the-html-5-specification/#about-browsers)),
+  which is bad for the standardization process.
+
+Sure, in retrospect MooTools did the wrong thing — but breaking the web
+doesn’t punish them, it punishes users. These users do not know what a moo
+tool is. Alternatively, we can find another solution, and users can continue
+to use the web. The choice is easy to make.
+
+## Does that mean we can never remove bad APIs from the Web Platform? {: #removing-apis }
+
+It depends. In rare cases we can get away with [removing bad features from
```

Comment:
Who is "we"? Chrome, Standards groups, Web platform at large?

Reply:
The Web Platform at large. Browsers do the telemetry investigation and try to remove it, and if they succeed, the standards bodies can update the spec accordingly. The WHATWG doc I link to explains it.

I’ll rephrase to avoid “we”.